hitman3.asl

```diff
@@ -1,58 +1,356 @@
+        {"LOCATION_WET", "level18"},
+        {"LOCATION_ELEGANT_LLAMA", "level19"},
+        {"LOCATION_ELEGANT", "level19"},
+        {"LOCATION_TRAPPED_WOLVERINE", "level20"},
+        {"LOCATION_TRAPPED", "level20"}
+    };
 }
 
-startup {
-    print("[hitman asl] startup");
-    vars.totalIGT = 0.0;
-    refreshRate = 30;
+start
+{
+    if (settings.StartEnabled && settings["autorestart"])
+    {
+        if (vars.currentMissionTime == 0)
+        {
+            string settingKey = string.Empty;
+            if (vars.locationDescriptor.TryGetValue(vars.currentMetadataLocation, out settingKey))
+            {
+                if (settings[settingKey])
+                {
+                    vars.totalIGT = (ulong)0;
+                    return true;
+                }
+            }
+        }
+    }
+    else
+    {
+        if (vars.totalIGT > 0)
+        {
+            vars.totalIGT = (ulong)0;
+        }
+    }
+
+    return false;
 }
 
-start {
-    if (vars.totalIGT > 0) {
-        print("[hitman asl] livesplit reset");
-        vars.totalIGT = 0.0;
+reset
+{
+    if (vars.currentMissionTime > 0)
+    {
+        vars.disableReset = false;
+    }
+
+    if (settings.ResetEnabled
+        && vars.disableReset == false
+        && settings["autorestart"]
+        && vars.currentGateExited == false
+        && vars.currentMissionTime == 0
+        && vars.currentHudMissionTimer > 0)
+    {
+        string settingKey = string.Empty;
+        if (vars.locationDescriptor.TryGetValue(vars.currentMetadataLocation, out settingKey))
+        {
+            if (settings[settingKey])
+            {
+                vars.disableReset = true;
+                return true;
+            }
+        }
     }
+
+    return false;
+}
+
+split
+{
+    //Split when exited mission
+    //
+    return settings.SplitEnabled && vars.currentHudMissionTimer > 0 && vars.oldGateExited == false && vars.currentGateExited == true;
 }
 
-isLoading {
+isLoading
+{
     return true;
 }
 
-gameTime {
-    if (current.ALT_IGT > old.ALT_IGT) {
-        vars.totalIGT += current.ALT_IGT - old.ALT_IGT;
+gameTime
+{
+    if (vars.currentTime > vars.oldTime && vars.currentMissionTime > 0
+        && vars.oldTime >= vars.currentMissionTime
+        && vars.currentGateExited == false)
+    {
+        vars.totalIGT += vars.currentTime - vars.oldTime;
     }
-    else if (current.CST > 0 && current.IGT > old.IGT && current.IGT - current.CST > 0.05) {
-        vars.totalIGT += current.IGT - old.IGT;
+
+    if (settings["useseconds"])
+    {
+        //Fix timer to seconds only
+        //
+        if ((vars.oldGateExited == false && vars.currentGateExited == true) ||
+            (vars.currentMissionTime == 0 && vars.currentHudMissionTimer > 0))
+        {
+            vars.totalIGT &= ~((ulong)0xFFFFF);
+        }
     }
-    return TimeSpan.FromSeconds(vars.totalIGT);
+    return TimeSpan.FromSeconds((double)vars.totalIGT * 0.00000095367432);
```

Comment:
At least add a comment on what the magic number here is doing -- though I think it would be easier to just have this stored as a double the whole time.

Reply:
Game does it, count time in binary nanoseconds and when SetText function is called for HudMissionTimer then convert it to seconds with that "magic" binary number

Reply:
Hm, interesting but odd. I guess add a comment about this

Reply:
Yeah, i should to make it more clear. I also was suprised about that while reversing